lib/Doctrine/ORM/Query/ResultSetMapping.php

```diff
@@ -583,5 +589,40 @@ public function addMetaResult($alias, $columnName, $fieldName, $isIdentifierColu
 
         return $this;
     }
-}
 
+    /**
+     * Retrieves the DBAL type name for the single identifier column of the root of a selection.
+     * Composite identifiers not supported!
```

Comment:
Is there any test covering exceptions when trying to paginate entities with composite identifiers? Perhaps it might be important to add one, especially because we're strengthening this restriction.

Reply:
I'll add one test specific to `ResultSetMapping`